types/instance.d.ts

```diff
@@ -370,6 +370,11 @@ export interface FastifyInstance<
    */
   setSerializerCompiler<T = FastifySchema>(schemaCompiler: FastifySerializerCompiler<T>): FastifyInstance<RawServer, RawRequest, RawReply, Logger>;
 
+  /**
+   * Set the schema controller for all routes.
+   */
+  setSchemaController(schemaControllerOpts: (payload: unknown, statusCode: number) => void): FastifyInstance<RawServer, RawRequest, RawReply, Logger>;
```

Comment:
Are you sure the type here is correct?

https://github.com/fastify/fastify/blob/1c8e4eda83cb0910cb03372a96d479b8226b1fb6/test/schema-feature.test.js#L1242-L1606

Reply:
I was in doubt with the return being void, I was hoping to find the Schema Controller type, but all that there is the Fastify Server Instance type. Perhaps i could make this a type using what we wave  in the line 133 too ? 
and then to test in this case it would return the schemaController ?

Reply:
Here is the actual implementation. 
https://github.com/fastify/fastify/blob/e632a9bc2f25091f533596a2f42de45963968e7d/fastify.js#L631-L639

The type should be

```ts
interface SchemaControllerOptions { 
   bucket?: (parentSchemas?: unknown) => { 
     addSchema(schema: unknown): FastifyInstance; 
     getSchema(schemaId: string): unknown; 
     getSchemas(): Record<string, unknown>; 
   }; 
   compilersFactory?: { 
     buildValidator?: ValidatorCompiler; 
     buildSerializer?: (externalSchemas: unknown, serializerOptsServerOption: FastifyServerOptions["serializerOpts"]) => FastifySerializerCompiler<unknown>; 
   }; 
 }
 
setSchemaController(schemaControllerOpts: SchemaControllerOptions ): FastifyInstance<RawServer, RawRequest, RawReply, Logger>;
```

Reply:
I've done the changes i think now the code is right but i wanted to make the Type tests better I looked on the web but i think I could do some more in the test part, also I wanted  to implement the compilersFactory type test but I really don't have any ideia how to do it :(  I could do the same way as I did on the bucket ? If you have any links to help me get better on this test part I would LOVE to learn more about this kind of testing